File: lists/linked_list.py
class node():
    def __init__(self, v=None):
        self.v = v
        self.next = None

class linked_list():
    def __init__(self):
        self.head = node()

    def append(self, v):
        if self.head.v == None:
            self.head.v = v
        else:
            current_node = self.head
            while current_node.next:
                current_node = current_node.next
            current_node.next = node(v)
    def prepend(self, v):
        if self.head.v == None:
            self.head.v = v
        else:
            current_node = node(v)
            current_node.next = self.head
            self.head = current_node
    def search(self, v):
        if self.head.v == None: return -1
        current_node = self.head
        index = 0
        while current_node.next:
            if current_node.v == v: return index
            current_node = current_node.next
            index += 1
        return -1
    def insert(self, v, index):
        if index == 0: return self.prepend(v)
        current_node = self.head
        while index:
            index -= 1
            vision_node = current_node.next.next
            if not vision_node: return -1
            current_node.next = node(v)
            current_node.next.next = vision_node
            return 0
        return -1
    def remove(self, v):
        if self.head.v == None: return -1
        if self.head.v == v:
            self.head = self.head.next
            return 0
        current_node = self.head
        while current_node.next:
            if current_node.next.v == v: 
                current_node.next = current_node.next.next
                return 0
            current_node = current_node.next
        return -1


# test
print('testing with: (1) --> (2) --> (3) --> (4)')
foo = linked_list()
foo.append(1)
foo.append(2)
foo.append(3)
foo.append(4)
foo.append(5)

File: cpp/src/linked_list/src/linked_list.cpp
#include "linked_list/linked_list.h"

namespace linked_list
{

template<typename T>
Node<T>::Node(T _value): value(_value) {}

template<typename T>
LinkedList<T>::LinkedList() {}

//template<typename T>
//LinkedList<T>::LinkedList(std::vector<T> data) {
	//for(auto e: data)
		//this->insert(e);
//}


template<typename T>
void LinkedList<T>::append(T& key) {
	if(!head){
		head = new Node<T>(key);
		return;
	}
	auto node = head;
	while(node->next)
		node = node->next;
	node->next = new Node<T>(key);
}

template<typename T>
Node<T>* LinkedList<T>::search(T& key) {
	if(!head)
		return nullptr;
	auto node = head;
	while(node){
		if(node->value == key)
			return node;
		node = node->next;
	}
	return nullptr;
}

template<typename T>
void LinkedList<T>::traverse() {
	if(!head)
		return;
	auto node = head;
	while(node){
		std::cout << "->(" + std::to_string(node->value) << ")";
		node = node->next;
	}
	std::cout << std::endl;
}

} // namespace linked_list

// only here for testing
int main(int argc, char const *argv[])
{
	if (argc < 2) {
		// report version
		std::cout << argv[0] << " Version " << "1.0.0" << ".\n";
		std::cout << "Usage: " << argv[0] << " <your name>" << std::endl;
		return 1;
	}

	int test = 69;
	auto linked = linked_list::LinkedList<int>();
	linked.append(test);
	test = 117;
	linked.append(test);
	test = 72;
	linked.append(test);
	linked.traverse();
	auto node = linked.search(test);
	if(node)
		std::cout << "found: " << std::to_string(node->value) << std::endl;
	//std::vector<int> test = {1, 2, 3, 4, 5};
	//auto list = linked_list::LinkedList<int>(test);
	//auto node = linked_list::Node<int>(69);
	//std::cout << argv[1] << " quiere mucho a su marion :(" << std::endl;
	//std::cout << "print node: " << std::to_string(node.value) << std::endl;
	return 0;
}
